backtracking/generate_parentheses.cpp

```diff
@@ -0,0 +1,113 @@
+/**
+ * @file
+ * @brief Well-formed [Generated
+ * Parentheses](https://leetcode.com/explore/interview/card/top-interview-questions-medium/109/backtracking/794/) with all combinations.
```

Comment:
@Panquesito7 should this not be in leetcode directory instead of in backtracking?

Reply:
It probably should, however, there are a few reasons:

1. There's no LeetCode folder yet (see #2450).
2. A few other algorithms belong to both the LeetCode section and their own section as well (Backtracking in this case).

I'm not sure if we should move it to the LeetCode directory, though.

Reply:
Ok, then its fine to keep it as is for now then 👍🏼

Reply:
We can ask them move it to leetcode after it's made

Reply:
If you create a LeetCode directory, I have a lot of problems that I can put

Reply:
> If you create a LeetCode directory, I have a lot of problems that I can put

there is already [one](https://github.com/TheAlgorithms/C-Plus-Plus/pull/2450) in the works